### DISCUSSION TITLE
[Help wanted] Is there a way to get page lang?

### DISCUSSION THREAD

**Discussion by Mister-Hope** [vuepress/core#47] [Q&A] [answered]
In V1, there is a internal variable called `page._computed.$lang` in `PageComputed`.

But V2, the `Page` in `extendsPage` only has `pathLocale`, and it's hard to get the actual lang from it.
(Currently the best way I found is to import `resolveSiteLocaleData` from `@vuepress/client` and called `resolveSiteLocaleData(app.options, page.pathLocale)` then check `result.locales.lang` )

I wonder if it there is another simple way.🤓

---

I think it's common for people to extend page with i18n related data, and that may need  `lang`. And we can't just regard `/` as english

**Comment by Mister-Hope** [comment]
Addtional: it's only possible to use `resolveSiteLocaleData ` with tools like rollup on node side, because `@vuepress/client` is complete output as esm syntax. @meteorlxy Any ideas with this one?

I don't think people should resolve page lang from locales in siteconfig and themeConfig and current page path by themseleves

**Comment by meteorlxy** [comment, marked as answer]
Page language is available in page data now 03bb09fd51aeaff56d26820a1401b87ea8bdeb38
